### ISSUE TITLE
Version 1.016000 has "Use of uninitialized value in sprintf" warnings

### ISSUE THREAD

**Issue by robrwo** [tobyink/p5-type-tiny#91]
Upgraded from version 1.014000 to Version 1.016000 and get the following warnings when running tests:
> Use of uninitialized value in sprintf at local/lib/perl5/Type/Params/Signature.pm line 183.

Downgrading removes the warnings.

**Comment by robrwo** [comment]
This was on Perl v5.34.0

**Comment by tobyink** [comment]
I don't get this warning myself, and I do my development on 5.34.0. This should hopefully fix any uninitialized warnings from that line though:

https://github.com/tobyink/p5-type-tiny/commit/515e56c832043d6899d55c6c3e6b9b4399ef1b8f

**Comment by djerius** [comment]
I'm getting similar messages under 5.36 using 1.016009, but only when using `wrap_subs:`

```
package Bug;

use v5.10;

use strict;
use warnings;

use Types::Standard -all;
use Type::Params -all;
sub query {
    # state $check = compile_named( a => Any );
    # @_ = $check->( @_ );
    use Data::Dumper;
    print Dumper( \@_ ), "\n";
}
wrap_subs query => compile_named( a => Any );

1;
```
```
% perl -I. -MBug -e 'Bug::query(a => q{b})'
Use of uninitialized value in sprintf at [...]/Type/Params/Signature.pm line 183.
Use of uninitialized value in sprintf at [...]/Type/Params/Signature.pm line 183.
$VAR1 = [
          {
            'a' => 'b'
          }
        ];
```
Switching over to the `state` version proceeds without complaint.

On occasion I'm also able to elicit the following errors:

```
Use of uninitialized value $package in concatenation (.) or string at [...]/Type/Params/Signature.pm line 551.
Use of uninitialized value $subname in concatenation (.) or string at [...]/Type/Params/Signature.pm line 551.
```

but I need to re-discover the combination that leads that error.

**Comment by djerius** [comment]
This will emit both errors:
```
package Bug;

use v5.36;

use Types::Standard -all;
use Type::Params -all;

wrap_subs query => compile_named_oo(
    uploads => Optional [HashRef],
);

sub query  {}

1;
```

```
% perl -I. -MBug -e 'Bug::query'
Use of uninitialized value in sprintf at [...]/Type/Params/Signature.pm line 183.
Use of uninitialized value in sprintf at [...]/Type/Params/Signature.pm line 183.
Use of uninitialized value $package in concatenation (.) or string at [...]/Type/Params/Signature.pm line 551.
Use of uninitialized value $subname in concatenation (.) or string at [...]/Type/Params/Signature.pm line 551.
```

**Comment by djerius** [comment]
I'm very sorry, I mistakenly tested this on 1.01600 instead of 1.016009.

The "uninitialized value in sprintf" error is gone, but the uninitialized values of `$package` and `$subname` is still there as of 1.016009.

In other words, the last example code I provided above emits this under 1.016009:

```
Use of uninitialized value $package in concatenation (.) or string at [...]/Type/Params/Signature.pm line 619.
Use of uninitialized value $subname in concatenation (.) or string at [...]/Type/Params/Signature.pm line 619.
```
Shall I open a new issue for this?

**Comment by tobyink** [comment]
I'll just reopen.

**Comment by tobyink** [comment]
Fixed in 1.016010.

**Comment by djerius** [comment]
Thanks!
